.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,25 @@
+---
+name: codeQL
+
+on:
+  push:
+    branches: [development/*, stabilization/*, hotfix/*]
+  pull_request:
+    branches: [development/*, stabilization/*, hotfix/*]
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Static analysis with CodeQL
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: javascript
+
+      - name: Build and analyze
+        uses: github/codeql-action/analyze@v2
```

Comment:
You're missing a set of steps here, 1, it's not javascript, it's java, 2 as it's java you need to build the project. 

use the default config of codeql with autobuild to experiment if it solves your issue

Reply:
I think it's better now

Reply:
Unfortunately, autobuild does not work, I will try to build by my own!

Reply:
I change the `autobuild` step to a build with Gradle :
```
- name: Build with Gradle
        run: |
          chmod +x gradlew
          ./gradlew build --no-daemon
```
I tested it on local but it does work because the creds `sonatypeUsername` & `sonatypePassword` are not given to the project - I pushed it anyways because I think, in prod, it'll be alright, tell me if I'm wrong!

Reply:
If autobuild didn't worked, have a look around you, there's already a ci for this project that is building it and executing some tests take inspiration from the existing methods.